### DISCUSSION TITLE
Out of memory error 0-16-3 beta

### DISCUSSION THREAD

**Discussion by dunkfan** [lightningnetwork/lnd#7792] [Troubleshooting] [answered]
For the second time 0-16-3 has died on me. No problems with 0-16.0.

Here's the output... A restart fixes it for a few days
No other process has problems.... 8GB Mem 2.3GB in use
Thanks

top output shortly before "the end"...

top - 22:13:39 up 6 days,  4:13,  2 users,  load average: 0.33, 0.53, 0.57
Tasks: 177 total,   1 running, 176 sleeping,   0 stopped,   0 zombie
%Cpu(s):  4.9 us,  0.8 sy,  0.0 ni, 93.5 id,  0.7 wa,  0.0 hi,  0.1 si,  0.0 st
MiB Mem :   7898.7 total,    203.5 free,   3176.9 used,   4518.4 buff/cache
MiB Swap:  58656.0 total,  58016.0 free,    640.0 used.   4418.8 avail Mem 

  PID USER      PR  NI    VIRT    RES    SHR S  %CPU  %MEM     TIME+ COMMAND                                                                              
  666 debian-+  20   0   86480  53000   6460 S   5.9   0.7 316:49.23 tor                                                                                  
 6388 pi        20   0 1260664 991008   6216 S   5.6  12.3 826:00.18 bitcoind                                                                             
23814 ln2       20   0 2399120   1.4g 125732 S   4.0  18.3 453:38.25 lnd   

2023-06-25 22:22:27.493 [INF] NTFN: New block: height=795914, sha=00000000000000000000391b09d376824104aa6269da64c7569672fd3165c2ec
2023-06-25 22:22:27.494 [INF] UTXN: Attempting to graduate height=795914: num_kids=0, num_babies=0
2023-06-25 22:22:55.857 [INF] DISC: GossipSyncer(0225e29bcf897f22d9f546ad71185f9519db538d484b2fd2ea2be02e20b5c7224b): filtering chan range: start_height=525999, num_blocks=2048
2023-06-25 22:22:55.858 [INF] DISC: GossipSyncer(0225e29bcf897f22d9f546ad71185f9519db538d484b2fd2ea2be02e20b5c7224b): sending final chan range chunk, size=57
runtime: out of memory: cannot allocate 620756992-byte block (1939865600 in use)
fatal error: out of memory

goroutine 359 [running]:
runtime.throw({0x192fe0c, 0xd})
	runtime/panic.go:1047 +0x4c fp=0x11fbbcc8 sp=0x11fbbcb4 pc=0x4df3c
runtime.(*mcache).allocLarge(0xb6ee2088, 0x24ec0000, 0x1)
	runtime/mcache.go:236 +0x1b8 fp=0x11fbbcf0 sp=0x11fbbcc8 pc=0x26a30
runtime.mallocgc(0x24ec0000, 0x1771130, 0x1)
	runtime/malloc.go:1053 +0x650 fp=0x11fbbd28 sp=0x11fbbcf0 pc=0x1ceb8
runtime.newarray(0x1771130, 0x110000)
	runtime/malloc.go:1276 +0x9c fp=0x11fbbd3c sp=0x11fbbd28 pc=0x1d548
runtime.makeBucketArray(0x16b6738, 0x14, 0x0)
	runtime/map.go:363 +0x17c fp=0x11fbbd58 sp=0x11fbbd3c pc=0x1e454
runtime.hashGrow(0x16b6738, 0x5adae40)
	runtime/map.go:1051 +0x8c fp=0x11fbbd7c sp=0x11fbbd58 pc=0x1ffc4
runtime.mapassign(0x16b6738, 0x5adae40, 0x11fbbe28)
	runtime/map.go:658 +0xec fp=0x11fbbdbc sp=0x11fbbd7c pc=0x1ec84
github.com/btcsuite/btcwallet/chain.(*cachedInputs).addInput(0xc5f0090, {{0x19, 0xee, 0x27, 0x68, 0xff, 0x9c, 0x57, 0xc9, 0x63, ...}, ...}, ...)
	github.com/btcsuite/btcwallet@v0.16.10-0.20230517173256-aa62c04afcdf/chain/mempool.go:64 +0x2bc fp=0x11fbbe20 sp=0x11fbbdbc pc=0x5a2544
github.com/btcsuite/btcwallet/chain.(*mempool).updateInputs(0x4dc45a0, 0xb01cc40)
	github.com/btcsuite/btcwallet@v0.16.10-0.20230517173256-aa62c04afcdf/chain/mempool.go:274 +0x1d8 fp=0x11fbbec0 sp=0x11fbbe20 pc=0x5a31d0
github.com/btcsuite/btcwallet/chain.(*mempool).add(0x4dc45a0, 0xb01cc40)
	github.com/btcsuite/btcwallet@v0.16.10-0.20230517173256-aa62c04afcdf/chain/mempool.go:179 +0x78 fp=0x11fbbf08 sp=0x11fbbec0 pc=0x5a2a90
github.com/btcsuite/btcwallet/chain.(*mempool).Add(0x4dc45a0, 0xb01cc40)
	github.com/btcsuite/btcwallet@v0.16.10-0.20230517173256-aa62c04afcdf/chain/mempool.go:137 +0x70 fp=0x11fbbf24 sp=0x11fbbf08 pc=0x5a285c
github.com/btcsuite/btcwallet/chain.(*bitcoindZMQEvents).txEventHandler(0x4dc45d0)
	github.com/btcsuite/btcwallet@v0.16.10-0.20230517173256-aa62c04afcdf/chain/bitcoind_zmq_events.go:362 +0x588 fp=0x11fbbfe4 sp=0x11fbbf24 pc=0x59eb6c
github.com/btcsuite/btcwallet/chain.(*bitcoindZMQEvents).Start.func2()
	github.com/btcsuite/btcwallet@v0.16.10-0.20230517173256-aa62c04afcdf/chain/bitcoind_zmq_events.go:145 +0x28 fp=0x11fbbfec sp=0x11fbbfe4 pc=0x59dac8
runtime.goexit()
	runtime/asm_arm.s:831 +0x4 fp=0x11fbbfec sp=0x11fbbfec pc=0x87ae4
created by github.com/btcsuite/btcwallet/chain.(*bitcoindZMQEvents).Start
	github.com/btcsuite/btcwallet@v0.16.10-0.20230517173256-aa62c04afcdf/chain/bitcoind_zmq_events.go:145 +0xdc

**Comment by guggero** [comment, marked as answer]
See https://github.com/lightningnetwork/lnd/issues/7785, will be fixed with `lnd 0.16.4` which should come out shortly.
